.changeset/wicked-terms-sneeze.md

```diff
@@ -0,0 +1,5 @@
+---
+'urql': minor
+---
+
+Leverage the new `use-sync-external-store` for `useQuery` to make the implementation more in line with React. Our current implementation works already with React Suspense, ... but having an implementation that coexists with React will be better for future changes
```

Comment:
```suggestion
Leverage the new `use-sync-external-store` package and `useSyncExternalStore` hook in `useQuery` implementation to bring the state synchronisation in React in line with React v18. While the current implementation works already with React Suspense and React Concurrent this will reduce the maintenance burden of our implementation and ensure certain guarantees so that React doesn't break us.
```